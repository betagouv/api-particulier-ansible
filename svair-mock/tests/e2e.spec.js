const request = require('supertest');
const app = require('../app');

describe('The mock application', () => {
    let server;

    it('renders a mocked html page when called with test inputs', () => {
        server = request(app)
            .post('/secavis/faces/commun/index.jsf')
            .send('j_id_7:j_id_l=Valider')
            .send('j_id_7_SUBMIT=1')
            .send('j_id_7:spi=1906299999002')
            .send('j_id_7:num_facture=1906299999002')
            .expect(200)
            .then(response => {
                expect(response.text).toMatchSnapshot();
            })

        return server;
    });

    afterAll(() => {
        server.stop();
    })
});
