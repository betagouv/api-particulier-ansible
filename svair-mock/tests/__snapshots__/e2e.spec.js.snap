// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The mock application renders a mocked html page when called with test inputs 1`] = `
"<!DOCTYPE html
	PUBLIC \\"-//W3C//DTD XHTML 1.0 Transitional//EN\\" \\"http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd\\">
<html xmlns=\\"http://www.w3.org/1999/xhtml\\">

<head>
	<meta http-equiv=\\"Content-type\\" content=\\"text/html; charset=UTF-8\\">

	<title>Impots.gouv.fr - Service de vérification en ligne des avis</title>
	<link href=\\"Impots.gouv.fr%20-%20Service%20de%20v%C3%A9rification%20en%20ligne%20des%20avis_files/style.css\\"
		rel=\\"styleSheet\\" type=\\"text/css\\">
</head>

<body>

	<div id=\\"conteneur\\">
		<div id=\\"barre_haut\\">
			<div style=\\"float: left;\\"><img
					src=\\"Impots.gouv.fr%20-%20Service%20de%20v%C3%A9rification%20en%20ligne%20des%20avis_files/bo_seule-2.gif\\"
					alt=\\"\\"></div>
			<div style=\\"float: right;\\"><img
					src=\\"Impots.gouv.fr%20-%20Service%20de%20v%C3%A9rification%20en%20ligne%20des%20avis_files/bo_seule-3.gif\\"
					alt=\\"\\"></div>
		</div>
		<div id=\\"principal\\">
			<div id=\\"nav_pro\\">
				<b>L'administration fiscale certifie l'authenticité du document présenté pour les données suivantes
					:</b>
			</div>

			<div class=\\"titre_affiche_avis\\">
				<span>Impôt 2019 sur les revenus de l'année 2018
				</span>
			</div>


			<table>
				<tbody>
					<tr>
						<td class=\\"label\\">
						</td>
						<td class=\\"label\\">Déclarant 1
						</td>
						<td class=\\"label\\">Déclarant 2
						</td>
					</tr>
					<tr>
						<td class=\\"labelPair\\">Nom
						</td>
						<td class=\\"labelPair\\">BERNARD
						</td>
						<td class=\\"labelPair\\">DUPONT
						</td>
					</tr>
					<tr>
						<td class=\\"labelImpair\\">Nom de naissance
						</td>
						<td class=\\"labelImpair\\">BERNARD
						</td>
						<td class=\\"labelImpair\\">DUPONT
						</td>
					</tr>
					<tr>
						<td class=\\"labelPair\\">Prénom(s)
						</td>
						<td class=\\"labelPair\\">Philippe
						</td>
						<td class=\\"labelPair\\">Nathalie
						</td>
					</tr>
					<tr>
						<td class=\\"labelImpair\\">Date de naissance
						</td>
						<td class=\\"labelImpair\\">22/03/1985
						</td>
						<td class=\\"labelImpair\\">01/12/1973
						</td>
					</tr>
					<tr>
						<td class=\\"labelPair\\">
							addresse déclarée au 1<sup>er</sup> janvier 2019
						</td>
						<td class=\\"labelPair\\">18 Rue du Général Potez
						</td>
						<td class=\\"labelPair\\">
						</td>
					</tr>
					<tr>
						<td class=\\"labelPair\\">
						</td>
						<td colspan=\\"2\\" class=\\"labelPair\\">62170 Montreuil
						</td>
					</tr>
					<tr>
						<td class=\\"espace\\"></td>
					</tr>
					<tr>
						<td class=\\"labelPair\\" colspan=\\"2\\">Date de mise en recouvrement de l'avis d'impôt
						</td>
						<td class=\\"textPair\\">10/10/2019
						</td>
					</tr>
					<tr>
						<td class=\\"labelImpair\\" colspan=\\"2\\">Date d'établissement
						</td>
						<td class=\\"textImpair\\">08/07/2019
						</td>
					</tr>
					<tr>
						<td class=\\"labelPair\\" colspan=\\"2\\">Nombre de part(s)
						</td>
						<td class=\\"textPair\\">2
						</td>
					</tr>
					<tr>
						<td class=\\"labelImpair\\" colspan=\\"2\\">Situation de famille
						</td>
						<td class=\\"textImpair\\">Marié(e)s
						</td>
					</tr>
					<tr>
						<td class=\\"labelPair\\" colspan=\\"2\\">Nombre de personne(s) à charge
						</td>
						<td class=\\"textPair\\">2
						</td>
					</tr>
					<tr>
						<td class=\\"labelImpair\\" colspan=\\"2\\">Revenu brut global
						</td>
						<td class=\\"textImpair\\">18 410 €
						</td>
					</tr>
					<tr>
						<td class=\\"labelPair\\" colspan=\\"2\\">Revenu imposable
						</td>
						<td class=\\"textPair\\">18 410 €
						</td>
					</tr>
					<tr>
						<td class=\\"labelImpair\\" colspan=\\"2\\">Impôt sur le revenu net avant corrections
						</td>
						<td class=\\"textImpair\\">2 165 €
						</td>
					</tr>
					<tr>
						<td class=\\"labelPair\\" colspan=\\"2\\">Montant de l'impôt
						</td>
						<td class=\\"textPair\\">2 165 €
						</td>
					</tr>
					<tr>
						<td class=\\"labelImpair\\" colspan=\\"2\\">Revenu fiscal de référence
						</td>
						<td class=\\"textImpair\\">18 410 €
						</td>
					</tr>

				</tbody>
			</table>

			<div id=\\"situationPartielle\\">
			</div>

			<div id=\\"erreurCorrectif\\">Ce document ne correspond pas à la situation la plus récente pour cet usager</div>

			<div id=\\"boutonsAvis\\">
				<a class=\\"nouvelle_recherche\\" href=\\"/secavis/\\">
					Nouvelle recherche
				</a>
			</div>

		</div>
	</div>

</body>

</html>"
`;
